sharding/nodesCoordinator/indexHashedNodesCoordinatorRegistry.go

```diff
@@ -70,17 +70,9 @@ func GetNodesCoordinatorRegistry(
 	}
 
 	for epoch := int(lastEpoch) - 1; epoch >= minEpoch; epoch-- {
-		ncInternalkey := append([]byte(common.NodesCoordinatorRegistryKeyPrefix), []byte(fmt.Sprint(epoch))...)
-		log.Debug("getting nodes coordinator config", "key", ncInternalkey)
-
-		epochConfigBytes, err := storer.SearchFirst(ncInternalkey)
+		err := setEpochConfigPerEpoch(epoch, epochsConfig, storer)
 		if err != nil {
-			return nil, err
-		}
-
-		err = updateEpochsConfig(epochsConfig, epochConfigBytes)
-		if err != nil {
-			return nil, err
+			log.Debug("failed to get nodes coordinator config for epoch", "epoch", epoch)
```

Comment:
this was considered a critical error (in the previous version of the function)

Reply:
later on we considered that if we found the configuration for the most recent epoch, it is fine to continue, even if for the previous one it was not found; we should not return err if we can continue; this could happen at the upgrade